extensions/src/Smithy.Identity.Abstractions/Identity/IIdentityResolver.cs

```diff
@@ -17,5 +20,11 @@ public interface IIdentityResolver
         /// If the identity cannot be resolved an exception will be thrown.
         /// </summary>
         BaseIdentity ResolveIdentity();
+
+        /// <summary>
+        /// Loads the customer's identity for this resolver. 
+        /// If the identity cannot be resolved an exception will be thrown.
```

Comment:
Can we say what exception will be thrown? I'm guessing we are going to get into issues with the exception being AWS exception but just want to ask.

Reply:
Yes, it's an Amazon exception although I noticed the fallback credentials factory currently uses `AmazonServiceException` but I used `AmazonClientException` here. They should be the same, right?

Reply:
Actually it looks like `FallbackCredentialsFactory` has two spots it throws `AmazonClientException` and one spot where throws `AmazonServiceException`. Using `AmazonClientException` makes more sense to me.